library/coretests/benches/ascii/eq_ignore_ascii_case.rs

```diff
@@ -0,0 +1,56 @@
+use test::Bencher;
+
+#[bench]
+fn bench_str_under_8_bytes_eq(b: &mut Bencher) {
+    let s = "foo";
+    let other = "FOo";
+    b.iter(|| {
+        assert!(s.eq_ignore_ascii_case(other));
+    })
+}
+
+#[bench]
+fn bench_str_of_8_bytes_eq(b: &mut Bencher) {
+    let s = "foobar78";
+    let other = "FOObAr78";
+    b.iter(|| {
+        assert!(s.eq_ignore_ascii_case(other));
+    })
+}
+
+#[bench]
+fn bench_str_17_bytes_eq(b: &mut Bencher) {
+    let s = "performance-criti";
+    let other = "performANce-cRIti";
+    b.iter(|| {
+        assert!(s.eq_ignore_ascii_case(other));
+    })
+}
+
+#[bench]
+fn bench_str_31_bytes_eq(b: &mut Bencher) {
+    let s = "foobarbazquux02foobarbazquux025";
+    let other = "fooBARbazQuuX02fooBARbazQuuX025";
+    b.iter(|| {
+        assert!(s.eq_ignore_ascii_case(other));
+    })
+}
+
+#[bench]
+fn bench_long_str_eq(b: &mut Bencher) {
+    let s = "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor \
```

Comment:
`library/coretests/benches/str/corpora.rs` has a bunch of test strings, including ascii.

Reply:
I looked for any other benches as reference, but these seem to be the first benches to check for equality between slices or strings. Every other ASCII/str bench is a validation or transform.

I can use corpora strings and make a runtime String for `other` if that's preferred.
```rust
use super::corpora::*;
let s = en::MEDIUM;
let other = s.to_ascii_uppercase();
```

Reply:
For benchmarking I don't think the strings actually need to have different cases. With branchless code it should be executing the same stuff whether they're bit-identical or just casefold-identical.